src/OrchardCore.Modules/OrchardCore.Recipes/RecipeSteps/RecipesStep.cs

```diff
@@ -27,25 +27,16 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
             }
 
             var step = context.Step.ToObject<InternalStep>();
-            var recipesDictionary = new Dictionary<string, IDictionary<string, RecipeDescriptor>>();
-            IList<RecipeDescriptor> innerRecipes = new List<RecipeDescriptor>();
+            var recipeCollections = await Task.WhenAll(_recipeHarvesters.Select(x => x.HarvestRecipesAsync()));
+            var recipes = recipeCollections.SelectMany(x => x).ToDictionary(x => x.Name);
 
+            IList<RecipeDescriptor> innerRecipes = new List<RecipeDescriptor>();
             foreach (var recipe in step.Values)
             {
-                IDictionary<string, RecipeDescriptor> recipes;
-
-                if (!recipesDictionary.TryGetValue(recipe.ExecutionId, out recipes))
-                {
-                    var recipeCollections = await Task.WhenAll(_recipeHarvesters.Select(x => x.HarvestRecipesAsync()));
-                    recipes = recipeCollections.SelectMany(x => x).ToDictionary(x => x.Name);
-                    recipesDictionary[recipe.ExecutionId] = recipes;
-                }
-
                 if (!recipes.ContainsKey(recipe.Name))
```

Comment:
Maybe use `if (recipes.TryGetValue(recipe.Name. out var recipe))` to prevent two lookups